src/dir_contents.ml

```diff
@@ -97,10 +97,10 @@ let modules_of_files ~dir ~files =
       (* we aren't using Filename.extension because we want to handle
          filenames such as foo.cppo.ml *)
       match String.lsplit2 fn ~on:'.' with
-      | Some (s, "ml" ) -> Left  (make_module OCaml  s fn)
-      | Some (s, "re" ) -> Left  (make_module Reason s fn)
-      | Some (s, "mli") -> Right (make_module OCaml  s fn)
-      | Some (s, "rei") -> Right (make_module Reason s fn)
+      | Some (s, "ml" ) -> Left  (make_module Dialect.ocaml  s fn)
+      | Some (s, "re" ) -> Left  (make_module Dialect.reason s fn)
+      | Some (s, "mli") -> Right (make_module Dialect.ocaml  s fn)
+      | Some (s, "rei") -> Right (make_module Dialect.reason s fn)
```

Comment:
I guess we should look in an `extension -> dialect` dictionary here?

Reply:
Indeed, I am looking at doing that now.

Reply:
This is done now.